From 901805febacfcc2dcf5c4179455c2f0c47e81768 Mon Sep 17 00:00:00 2001
From: David Daney <david.daney@cavium.com>
Date: Mon, 20 May 2019 22:29:00 +0200
Subject: [PATCH 214/345] MIPS: OCTEON: Increase NR_IRQS for CONFIG_NUMA.

MSI requires 256 irqs per node, so 511 is not enough for a two node
system.

Signed-off-by: David Daney <david.daney@cavium.com>
---
 arch/mips/include/asm/mach-cavium-octeon/irq.h | 11 ++++++++---
 1 file changed, 8 insertions(+), 3 deletions(-)

diff --git a/arch/mips/include/asm/mach-cavium-octeon/irq.h b/arch/mips/include/asm/mach-cavium-octeon/irq.h
index bf0002b85aff..52b82c96ebb6 100644
--- a/arch/mips/include/asm/mach-cavium-octeon/irq.h
+++ b/arch/mips/include/asm/mach-cavium-octeon/irq.h
@@ -8,7 +8,13 @@
 #ifndef __OCTEON_IRQ_H__
 #define __OCTEON_IRQ_H__
 
-#define NR_IRQS OCTEON_IRQ_LAST
+#ifdef CONFIG_NUMA
+	/* We need 256 per node for MSI */
+#define NR_IRQS 767
+#else
+#define NR_IRQS 511
+#endif
+
 #define MIPS_CPU_IRQ_BASE OCTEON_IRQ_SW0
 
 /*
@@ -53,8 +59,7 @@ enum octeon_irq {
 	OCTEON_IRQ_SRIO0,
 	OCTEON_IRQ_SRIO1,
 	OCTEON_IRQ_SRIO2,
-	OCTEON_IRQ_SRIO3,
-	OCTEON_IRQ_LAST = 511
+	OCTEON_IRQ_SRIO3
 };
 
 #endif
-- 
2.25.1

